Classes/common/HTTP/CDTURLSessionInterceptorContext.m

```diff
@@ -0,0 +1,35 @@
+//
+//  CDTURLSEssionFilterContext.m
+//  
+//
+//  Created by Rhys Short on 17/08/2015.
+//
+//
+
+#import "CDTURLSEssionInterceptorContext.h"
+
+@implementation CDTURLSessionInterceptorContext
+
+//turn off the nonnull warning, we are passing null to cause a failure,
+// trying to use init should cause a compiler error anyway
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wnonnull"
+-(instancetype) init {
+    return [self initWithRequest:nil];
```

Comment:
OK, so with regard to the above comment, I see you have marked `init` with `UNAVAILABLE_ATTRIBUTE` - so why give an implementation? And if the compiler requires you to give an implementation just `return nil` - then you won't need to mess about diagnostic pragmas for a method which never gets called.